From 38ac90768c12d69f375d2ad5a46c8d9168e692d6 Mon Sep 17 00:00:00 2001
From: John Greene <jogreene@redhat.com>
Date: Thu, 14 Aug 2014 14:42:22 -0400
Subject: [ethernet] e1000e: use SPEED_UNKNOWN and DUPLEX_UNKNOWN when appropriate

Message-id: <1408027348-9610-29-git-send-email-jogreene@redhat.com>
Patchwork-id: 87727
O-Subject: [RHEL7.1 PATCH 28/34] net: use SPEED_UNKNOWN and DUPLEX_UNKNOWN when appropriate
Bugzilla: 1091119
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

[redhat e1000e only]
commit 537fae0101c8853abb52136788173dde74b9d1e9
Author: Jiri Pirko <jiri@resnulli.us>
Date:   Fri Jun 6 14:17:00 2014 +0200

    net: use SPEED_UNKNOWN and DUPLEX_UNKNOWN when appropriate

Signed-off-by: Jiri Pirko <jiri@resnulli.us>
Signed-off-by: David S. Miller <davem@davemloft.net>

BZ: 1091119
Testing: Acceptable, see BZ
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7834849
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/e1000e/ethtool.c b/drivers/net/ethernet/intel/e1000e/ethtool.c
index cad250b..b04bad9 100644
--- a/drivers/net/ethernet/intel/e1000e/ethtool.c
+++ b/drivers/net/ethernet/intel/e1000e/ethtool.c
@@ -159,8 +159,8 @@ static int e1000_get_settings(struct net_device *netdev,
   ecmd->transceiver = XCVR_EXTERNAL;
  }
 
- speed = -1;
- ecmd->duplex = -1;
+ speed = SPEED_UNKNOWN;
+ ecmd->duplex = DUPLEX_UNKNOWN;
 
  if (netif_running(netdev)) {
   if (netif_carrier_ok(netdev)) {
-- 
1.7.1